delivery_carrier_package_measure_required/tests/test_delivery_carrier_package_measure_require.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+
+from odoo.exceptions import ValidationError
+
+from odoo.addons.stock.tests.test_packing import TestPackingCommon
+
+
+class TestDeliveryCarrierPackageMeasureRequire(TestPackingCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.normal_carrier = cls.env.ref("delivery.free_delivery_carrier")
+        cls.uom_kg = cls.env.ref("uom.product_uom_kgm")
+        cls.product_aw = cls.env["product.product"].create(
+            {
+                "name": "Product AW",
+                "type": "product",
+                "weight": 2.4,
+                "uom_id": cls.uom_kg.id,
+                "uom_po_id": cls.uom_kg.id,
+            }
+        )
+        cls.packaging = cls.env["stock.package.type"].create(
+            {"name": "Test Delivery Packaging"}
+        )
+        cls.env["stock.quant"]._update_available_quantity(
+            cls.product_aw, cls.stock_location, 20.0
+        )
+        cls.pick = cls.env["stock.picking"].create(
+            {
+                "partner_id": cls.env["res.partner"].create({"name": "A partner"}).id,
+                "picking_type_id": cls.warehouse.out_type_id.id,
+                "location_id": cls.stock_location.id,
+                "location_dest_id": cls.customer_location.id,
+                "carrier_id": cls.normal_carrier.id,
+            }
+        )
+        cls.env["stock.move.line"].create(
+            {
+                "product_id": cls.product_aw.id,
+                "product_uom_id": cls.uom_kg.id,
+                "picking_id": cls.pick.id,
+                "qty_done": 5,
+                "location_id": cls.stock_location.id,
+                "location_dest_id": cls.customer_location.id,
+            }
+        )
+        cls.pick.action_confirm()
```

Comment:
I know that this is a migration and comes from there, but to respect the stock flow in Odoo, it's better to create a `stock.move` (instead of a `stock.move.line`) directly associated with `stock.picking`, and then force availability using the 'action_assign' method. If there is availability, the `stock.move.line` is created, and we can modify its `qty_done`.

Reply:
done